Mathlib/Algebra/GroupWithZero/Range.lean

```diff
@@ -111,8 +111,55 @@ lemma mem_valueGroup {b : Bˣ} (hb : b.1 ∈ range f) : b ∈ valueGroup f := by
 lemma inv_mem_valueGroup {b : Bˣ} (hb : b.1 ∈ range f) : b⁻¹ ∈ valueGroup f :=
   Subgroup.inv_mem _ (mem_valueGroup f hb)
 
-end MonoidWithZero
+end MonoidWithZeroHom
+
+noncomputable section Restrict
+
+variable [MonoidWithZero A] [GroupWithZero B] [MonoidWithZeroHomClass F A B] {f}
+
+open Classical in
+/-- The inclusion of `valueGroup₀ f` into `B` as a multiplicative homomorphism. -/
+def valueGroup₀_MulWithZeroEmbedding : valueGroup₀ f →*₀ B :=
+  MonoidWithZeroHom.comp (WithZero.withZeroUnitsEquiv (G := B))
+    <| WithZero.map' (valueGroup f).subtype
 
+
+variable (f) in
+open Classical in
```

Comment:
```suggestion
```suggestion
variable [MonoidWithZero A] [GroupWithZero B] [DecidablePred fun b : B ↦ b = 0]
  [MonoidWithZeroHomClass F A B] {f}

/-- The inclusion of `valueGroup₀ f` into `B` as a multiplicative homomorphism. -/
def valueGroup₀_MulWithZeroEmbedding : valueGroup₀ f →*₀ B :=
  MonoidWithZeroHom.comp (WithZero.withZeroUnitsEquiv (G := B))
    <| WithZero.map' (valueGroup f).subtype


variable (f) in
```
one can avoid classical. decidable equality to zero is present in any WithZero X. And will also work better for obviously decidable Bs like Zm0.